### ISSUE TITLE
Tests pollute database

### ISSUE THREAD

**Issue by markmacgillivray** [ourresearch/total-impact-core#66]
We have tests now that write to the main database, and some that create new databases but never delete them. It may be that this is caused by my change of removing the default_settings, however, the original dao and api code and tests managed to run successfully in a test database and clean up after itself, so I do not know why what we have now cannot do that.

**Comment by markmacgillivray** [comment]
Have set test_api to write to ti_test as it used to but setting it to delete in the teardown causes document update conflicts. I do not yet know why this is, however I know that this previously worked fine when managed at the class level with ordered tests.

**Comment by jasonpriem** [comment]
The leftover databases at the end are meant to be a feature, not a bug. I've found that being able to look at the state of the database after tests to be a really useful diagnostic. You just have to be sure to delete the databases in the setUp methods.

Of course, we've also got the problem of all this setup/teardown of DB for every test taking a ton of time. That ain't good. But I reckon that goes on its own ticket if it becomes enough of a problem.

I'm ready to close this one if that's ok with you. Or we can continue discussion in this thread.
